src/ParallelAlgorithms/Events/ObserveAtExtremum.hpp

```diff
@@ -0,0 +1,412 @@
+            typename Metavariables, size_t VolumeDim,
+            typename ParallelComponent>
+  void operator()(const typename ObservationValueTag::type& observation_value,
+                  const ObservationBox<ComputeTagsList, DataBoxType>& box,
+                  Parallel::GlobalCache<Metavariables>& cache,
+                  const ElementId<VolumeDim>& array_index,
+                  const ParallelComponent* const /*meta*/) const;
+
+  using observation_registration_tags = tmpl::list<::Tags::DataBox>;
+
+  template <typename DbTagsList>
+  std::optional<
+      std::pair<observers::TypeOfObservation, observers::ObservationKey>>
+  get_observation_type_and_key_for_registration(
+      const db::DataBox<DbTagsList>& box) const {
+    const std::optional<std::string> section_observation_key =
+        observers::get_section_observation_key<ArraySectionIdTag>(box);
+    if (not section_observation_key.has_value()) {
+      return std::nullopt;
+    }
+    return {{observers::TypeOfObservation::Reduction,
+             observers::ObservationKey(
+                 subfile_path_ + section_observation_key.value() + ".dat")}};
+  }
+
+  using is_ready_argument_tags = tmpl::list<>;
+
+  template <typename Metavariables, typename ArrayIndex, typename Component>
+  bool is_ready(Parallel::GlobalCache<Metavariables>& /*cache*/,
+                const ArrayIndex& /*array_index*/,
+                const Component* const /*meta*/) const {
+    return true;
+  }
+
+  bool needs_evolved_variables() const override { return true; }
+
+  // NOLINTNEXTLINE(google-runtime-references)
+  void pup(PUP::er& p) override;
+
+ private:
+  std::string subfile_path_;
+  std::string scalar_name_;
+  std::string extremum_type_;
+  std::vector<std::string> additional_tensor_names_{};
+};
+/// @}
+
+/// \cond
+template <typename ObservationValueTag, typename... ObservableTensorTags,
+          typename... NonTensorComputeTags, typename ArraySectionIdTag>
+ObserveAtExtremum<ObservationValueTag, tmpl::list<ObservableTensorTags...>,
+                  tmpl::list<NonTensorComputeTags...>,
+                  ArraySectionIdTag>::ObserveAtExtremum(CkMigrateMessage* msg)
+    : Event(msg) {}
+
+template <typename ObservationValueTag, typename... ObservableTensorTags,
+          typename... NonTensorComputeTags, typename ArraySectionIdTag>
+ObserveAtExtremum<ObservationValueTag, tmpl::list<ObservableTensorTags...>,
+                  tmpl::list<NonTensorComputeTags...>, ArraySectionIdTag>::
+    ObserveAtExtremum(const std::string& subfile_name,
+                      const ObserveTensors& observe_tensors)
+    : subfile_path_("/" + subfile_name),
+      scalar_name_(observe_tensors.scalar_name),
+      extremum_type_(observe_tensors.extremum_type) {
+  additional_tensor_names_.reserve(observe_tensors.additional_data.size());
+  for (const auto& additional_tensor_name : observe_tensors.additional_data) {
+    additional_tensor_names_.push_back(additional_tensor_name);
+  }
```

Comment:
Move `observe_tensors.additional_data` into `additional_tensor_names_` instead of copying the elements.